File: Sorting/lab6(sorting in python)/insertion.py
import time
import random

def insertion_sort(arr):
    # Loop for passes
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i - 1
        # Loop for each pass
        while j >= 0 and arr[j] > temp:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = temp

def display(arr):
    print(" ".join(map(str, arr)))

def main():
    n = int(input("Enter n: "))
    arr = [random.randint(0, 100000) for _ in range(n)]

    display(arr)
    start = time.time()
    insertion_sort(arr)
    end = time.time()
    display(arr)

    print(f"The time taken is {end - start:.6f} seconds")

if __name__ == "__main__":
    main()



# 10 = The time taken is 0.000015  seconds
# 100 = The time taken is 0.000126 seconds
# 1000 = The time taken is 0.021856 seconds
# 10000 = The time taken is 1.213831 seconds
# 100000 = The time taken is 151.065459 seconds

File: Sorting/lab5(Sorting in C)(class)/insertion.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void insertionSort(int A[], int n)
{
    int temp, j;
    // Loop for passes
    for (int i = 1; i <= n-1; i++)
    {
        temp = A[i];
        j = i-1;
        // Loop for each pass
        while(j>=0 && A[j] > temp){
            A[j+1] = A[j];
            j--;
        }
        A[j+1] = temp;
    }
}

void display(int A[], int n)
{
    int i;
    for (i = 0; i < n; i++)
    {
        printf("%d  ", A[i]);
    }
    printf("\n");
}

int main()
{
    int A[100000], n, i;
    srand(time(NULL));
    clock_t start, end, diff;
    printf("Enter n:");
    scanf("%d", &n);
    for (i = 0; i < n; i++)
    {
        A[i] = rand();
    }
    display(A, n);
    start = clock();
    insertionSort(A, n);
    end = clock();
    display(A, n);
    diff = end - start;
    printf("the time taken is %f seconds\n", (float)diff / CLOCKS_PER_SEC);
    return 0;
}


// 10 = the time taken is 0.000000 seconds
// 100 = the time taken is 0.000000 seconds
// 1000 = the time taken is 0.000000 seconds
// 10000 = the time taken is 0.028000 seconds
// 100000 = the time taken is 2.721000 seconds